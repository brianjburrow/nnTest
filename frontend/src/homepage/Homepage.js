import React, { useContext } from "react";
import { Link } from "react-router-dom";
import "./Homepage.css";
import UserContext from "../auth/UserContext";

/** Homepage of site.
 *
 * Shows welcome message or login/register buttons.
 *
 * Routed at /
 *
 * Routes -> Homepage
 */

function Homepage() {
    const user = useContext(UserContext);
    console.debug("Homepage", "currentUser=", user);

    return (
        <div className="Homepage">
            <div className="container text-center">
                <h1 className="mb-4 font-weight-bold">nnTest</h1>
                <p className="lead">nnTest is a way to test out LSTM structures on forecasting tasks.</p>
                {user && user.currentUser
                    ? <h2>
                        Welcome Back, {user.currentUser.firstName || user.currentUser.username}!
                    </h2>
                    : (
                        <p>
                            <Link className="btn btn-primary font-weight-bold mr-3"
                                to="/login">
                                Log in
                            </Link>
                            <Link className="btn btn-primary font-weight-bold"
                                to="/signup">
                                Sign up
                            </Link>
                        </p>
                    )}
            </div>
        </div>
    );
}

export default Homepage;
