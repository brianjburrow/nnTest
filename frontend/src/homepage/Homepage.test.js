// tests based on Jobly App (Springboard)

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Homepage from "./Homepage";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
    render(<Homepage />);
});

it("matches snapshot when logged in", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider>
            <Homepage />
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider currentUser={null}>
            <Homepage />
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  