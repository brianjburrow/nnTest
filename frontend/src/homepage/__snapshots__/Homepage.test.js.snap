// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        nnTest
      </h1>
      <p
        class="lead"
      >
        nnTest is a way to test out LSTM structures on forecasting tasks.
      </p>
      <h2>
        Welcome Back, testfirst!
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        nnTest
      </h1>
      <p
        class="lead"
      >
        nnTest is a way to test out LSTM structures on forecasting tasks.
      </p>
      <p>
        <a
          class="btn btn-primary font-weight-bold mr-3"
          href="/login"
        >
          Log in
        </a>
        <a
          class="btn btn-primary font-weight-bold"
          href="/signup"
        >
          Sign up
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
